bcf_sr_sort.c

```diff
@@ -575,6 +572,19 @@ int bcf_sr_sort_next(bcf_srs_t *readers, sr_sort_t *srt, const char *chr, int mi
         srt->nsr = readers->nreaders;
         srt->chr = NULL;
     }
+    if ( srt->nactive == 1 )
+    {
+        if ( readers->nreaders>1 )
+            memset(readers->has_line, 0, readers->nreaders*sizeof(*readers->has_line));
+        bcf_sr_t *reader = &readers->readers[srt->active[0]];
+        assert( reader->buffer[1]->pos==min_pos );
+        bcf1_t *tmp = reader->buffer[0];
+        for (j=1; j<=reader->nbuffer; j++) reader->buffer[j-1] = reader->buffer[j];
```

Comment:
How large is nbuffer expected to be?  This is probably more efficient, as well as clearer, written as

```
memmove(&reader->buffer[j-1], &reader->buffer[j], reader->nbuffer);
```

I haven't profiled it, but if it's called many times over it's more efficient still using a cyclic ring-buffer of some sort so there is no data-copying required.  It may be over-optimisation though so probably not worth it and I expect you've profiled this already to spot the need for this PR.

Reply:
The difference with `memmove` will not be measurable, the size of these buffers is very small, typically 1-2, rarely bigger than 10.  The bottleneck is `bcf_sr_sort_set`, I may need to rethink the strategy and the use of hashes there.

Thank you for looking at this.

Reply:
Fair enough.  Thanks.